flight/core/Dispatcher.php

```diff
@@ -66,20 +76,28 @@ public function run(string $name, array $params = [])
     /**
      * Assigns a callback to an event.
      *
-     * @param string   $name     Event name
-     * @param callable $callback Callback function
+     * @param string $name Event name
+     * @param Closure(): (void|mixed) $callback Callback function
+     *
+     * @return $this
      */
-    public function set(string $name, callable $callback): void
+    public function set(string $name, callable $callback): self
     {
+        if ($this->get($name) !== null) {
```

Comment:
So I see what you're doing here, but this may actually cause other peoples older code to break. They might actually have bad code in place and this will force an error to be thrown and break their app, which is breaking backwards compatibility

Reply:
I'll take a look at the scenarios that could throw an exception. Could you provide a use case that would fail now due to an exception and not before? Could you provide a use case that would fail now due to an exception and not before?

Reply:
I guess I'm thinking that in someone's code they might have something like `Dispatcher->set('someMethod')` and then in a different part of their code, like a controller, they might override that method with the same name to do a different behavior.

I'm not agreeing that that is a good way to go, but I'm just saying it's possible. Going through all the issues and cleaning them up showed that some people have really bizarre work arounds for things like pseudo-middleware and whatnot.

Reply:
Well, to start, it's `(new Dispatcher)->set('someMethod')` instead of Dispatcher::set('someMethod')`.

So the only case when a method'll be overriden an therefore thrown a E_NOTICE is when the controller use the same object that it's used in other app parts

Reply:
I prefer to throw an E_NOTICE instead of an Exception to not make a breaking change